clang/lib/CConv/ProgramVar.cpp

```diff
@@ -32,7 +32,8 @@ const StructScope *StructScope::getStructScope(std::string StName) {
   if (AllStScopes.find(TmpS) == AllStScopes.end()) {
     AllStScopes.insert(TmpS);
   }
-  return &(*AllStScopes.find(TmpS));
+  const auto &SS = *AllStScopes.find(TmpS);
+  return &SS;
```

Comment:
Returning `&x` where `x` is a local variable is bad, isn't it? Same with what's below.

Reply:
Its a reference. So, returning `&SS` where `SS` is a reference is as good as returning a pointer.